indra/newview/llfetchedgltfmaterial.cpp

```diff
@@ -175,3 +177,55 @@ void LLFetchedGLTFMaterial::materialComplete()
     materialCompleteCallbacks.clear();
     materialCompleteCallbacks.shrink_to_fit();
 }
+
+LLPointer<LLViewerFetchedTexture> LLFetchedGLTFMaterial::getUITexture()
+{
+    if (mFetching)
+    {
+        return nullptr;
+    }
+
+    auto fetch_texture_for_ui = [](LLPointer<LLViewerFetchedTexture>& img, const LLUUID& id)
+    {
+        if (id.notNull())
+        {
+            if (LLAvatarAppearanceDefines::LLAvatarAppearanceDictionary::isBakedImageId(id))
+            {
+                LLViewerObject* obj = LLSelectMgr::getInstance()->getSelection()->getFirstObject();
```

Comment:
This looks wrong, relying onto selection manager when we are previewing materials will bite us in the rear later.

Baked images are limited in number and are shared between all objects. Shouln't checking LLAvatarAppearanceDictionary::isBakedImageId then assetIdToBakedTextureIndex be enough? If it's a magic id simply assume that this is an object (especially since this particular material can be applied to multiple objects and parameters like 'animated' can change) and baked image should be displayed as baked, we do the same in texture picker.